contrib/planb-objsync.py

```diff
@@ -1215,8 +1474,8 @@ def process_record(self, record, container, dst_path):
         # If the length is the same, we might be looking at an existing object.
         # Try to not download it again if possible.
         local_size = os.stat(dst_path).st_size  # should exist and succeed
-        if SwiftHeaders.contentlength(obj_stat) == local_size:
-            if SwiftHeaders.is_dynamic_large_object(obj_stat):
+        if ObjectHeaders.contentlength(obj_stat) == local_size:
+            if self._client.is_dynamic_large_object(obj_stat):
                 # We cannot check the md5sum in the DLO case. The etag is built
                 # up from the base16 md5sums of the individual parts, i.e.:
                 # etag = md5sum.hexdigest(
```

Comment:
Do you know whether this conclusion is correct for S3?

Reply:
Yes, the Etag calculation is similar to the DLO md5 with swiftclient.
But with S3 it also appends the total number of parts.
Note that the etag has already been validated at this point in code.
```
    def is_valid_etag(self, etag):                                              
        '''                                                                     
        Validate the etag value for the client.                                 
        '''                                                                     
        # The S3 multipart etag is a md5 hash with a parts count.               
        # {md5(''.join(md5(part) for part in parts))}-{len(parts)}              
        if super().is_valid_etag(etag):                                         
            return True                                                         
        try:                                                                    
            etag, parts = etag.split('-')                                       
        except (AttributeError, ValueError):                                    
            return False                                                        
        else:                                                                   
            return bool(super().is_valid_etag(etag) and parts.isdigit())
```
So the dlo check is simplified to look for the parts separator.
```
    def is_dynamic_large_object(self, headers):
        return bool('-' in headers.get('etag', ''))
```